src/App.tsx

```diff
@@ -1,61 +1,157 @@
+  const [titleError, setTitleError] = useState(false);
+  const [selectError, setSelectError] = useState(false);
+
+  const handleChange = (inputValue: string) => {
+    const reg = new RegExp('[a-zA-Zа-яА-Я\\d ]+');
+    const letter = inputValue[inputValue.length - 1];
+
+    setTitleError(false);
+
+    if (letter !== undefined) {
+      if (letter.search(reg) !== -1) {
+        setTitleName(inputValue);
+      }
+    } else {
+      setTitleName('');
+    }
+  };
+
+  const chooseSelect = (selectedValue: string) => {
+    setSelectError(false);
+    setUserName(selectedValue);
+  };
+
+  const clearingForm = () => {
+    setUserName('0');
+    setTitleName('');
+  };
+
+  const addTodo = () => {
+    const arrayId = (
+      [...todos]
+        .sort((a, b) => (b.id - a.id))[0].id + 1
+    );
+
+    const newTodo = {
+      id: arrayId,
+      title: titleName,
+      completed: false,
+      userId: +userName,
+      user: getUser(+userName),
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const onSubmit = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    if (!titleName.trim().length) {
+      setTitleError(true);
+    }
+
+    if (userName === '0') {
+      setSelectError(true);
+    }
+
+    if (userName !== '0' && titleName.trim().length > 0) {
+      addTodo();
+      clearingForm();
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            {'Title: '}
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={titleName}
+              onChange={(event) => {
+                handleChange(event.target.value);
+              }}
+            />
+          </label>
+
+          { titleError && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            {'User: '}
+            <select
+              data-cy="userSelect"
+              value={userName}
+              onChange={(select) => {
+                chooseSelect(select.target.value);
+              }}
+            >
+              <option value="0" disabled>Choose a user</option>
+              {usersFromServer.map((user) => {
+                const {
+                  id,
+                  name,
+                } = user;
+
+                return (
+                  <option
+                    value={id}
+                    key={id}
+                  >
+                    {name}
+                  </option>
+                );
+              })}
```

Comment:
it can be simplified
```suggestion
              {usersFromServer.map((user) => (
                <option value={user.id} key={user.id}>
                  {name}
                </option>
              )}
```